src/vanilla/Templates/MethodJsonPipelineTemplate.cshtml

```diff
@@ -10,34 +11,62 @@ try {
   @(Model.ConstructRequestBodyMapper)
 </text>
 }
-  operationRes = await client.sendOperationRequest(httpRequest, {
-@if (Model.RequestBody == null)
+  operationRes = await client.sendOperationRequest(
+    httpRequest,
+    {
+      arguments: {
+@foreach (Parameter parameter in Model.LogicalParameters.Where(p => p.Location == ParameterLocation.Header))
 {
 <text>
-    httpMethod: "@(Model.HttpMethod.ToString().ToUpper())"
+        "@(parameter.Name)": @(parameter.Name),
 </text>
 }
-else
+      },
+      customHeaders: new msRest.HttpHeaders(options && options.customHeaders)
+    },
+    {
+      httpMethod: "@(Model.HttpMethod.ToString().ToUpper())",
+@if (Model.LogicalParameters.Any(p => p.Location == ParameterLocation.Header))
 {
 <text>
-    httpMethod: "@(Model.HttpMethod.ToString().ToUpper())",
-    requestBodyMapper: requestModelMapper,
+      headerParameters: [
+@foreach (ParameterTS parameter in Model.LogicalParameters.Where(p => p.Location == ParameterLocation.Header))
+{
+<text>
+        {
+          parameterName: "@(parameter.Name)",
+@if (parameter.Name != parameter.SerializedName)
```

Comment:
Can we fail fast in these kinds of cases? (a header parameter that doesn't specify the header name seems like a Swagger problem.)

Reply:
This particular line isn't an error case, it's an optimization. In this scenario I'm saying that if the name of the parameter is the same as the name of the header, then don't add the header name because we can just use the parameter name for both.

Reply:
Yeah, just totally misread this yesterday 🤦‍♂️